sdcm/nemesis.py

```diff
@@ -2043,6 +2045,70 @@ def modify_table_speculative_retry(self):
                    "'%sms'" % random.randint(1, 1000))
         self._modify_table_property(name="speculative_retry", val=random.choice(options))
 
+    def modify_table_twcs_window_size(self):
+        """ Change window size for tables with TWCS
+
+            After window size of TWCS changed, tables should be
+            reshaped. Process should not bring write amplification
+            if size of sstables in timewindow is differs significantly
+        """
+
+        def get_new_window_settings(current_size: int, current_unit: str) -> Tuple[int, str]:
+
+            if current_unit == "DAYS":
+                current_size = current_size + 1
+            elif current_unit == "HOURS":
+                if current_size // 24 > 2:
+                    current_unit = "DAYS"
+                    current_size = 3
+                else:
+                    current_size += 10
+            else:
+                if current_size // 60 > 10:
+                    current_unit = "HOURS"
+                    current_size = 11
+                else:
+                    current_size += 35
+            return current_size, current_unit
+
+        ks_cs_with_twcs = self.cluster.get_all_tables_with_twcs(self.target_node)
+        self.log.debug("All tables with TWCS %s", ks_cs_with_twcs)
+
+        if not ks_cs_with_twcs:
+            raise UnsupportedNemesis('No table found with TWCS')
+
+        ks_cs = random.choice(ks_cs_with_twcs)
+        new_size, new_unit = get_new_window_settings(
+            int(ks_cs["compaction"]["compaction_window_size"]),
+            ks_cs["compaction"]["compaction_window_unit"])
+
+        ks_cs["compaction"]["compaction_window_size"] = new_size
+        ks_cs["compaction"]["compaction_window_unit"] = new_unit
+
+        self.log.debug("New TWCS settings %s", str(ks_cs["compaction"]))
+        self._modify_table_property(name="compaction", val=ks_cs["compaction"], keyspace_table=ks_cs["name"])
+
+        self.cluster.wait_for_schema_agreement()
+        self.target_node.stop_scylla()
```

Comment:
Do we want to have a sleep here for a while?
Maybe to have some of the sstables to be written with the new properties, so there will be a mixture of the old sstables and the new sstables.
If you do wait, let's make it as a variable based on test duration or something like that.

Reply:
Added